### PR TITLE
The Infiltrator Sneaksuit Bundle: Armor for Cheap

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Gosh this PR has really changed a lot of over time.

The Infiltrator Bundle is a sneaksuit set that allows the wearer the benefits of a decent armor set at a relatively low price of 3TC. The armor vest paired with the sneaksuit will equal a syndicate hardsuit, but only on the chest. It also carries none of the benefits of the hardsuit, such as EVA. 

The bundle comes complete with the following items:

**Infiltrator Helmet:** It's a insidious helmet. Has flashproofing.
**Infiltrator Vest:** It's roughly equal to an armor vest, but a lot wearker against lasers and disablers.
**Blood-red Sneaksuit:** It's the highest armored undersuit in the game, at 10 across the board.
**Infiltrator Balaclava:** Equal to an explorer mask in armor, and has a diagnostic hud built in. It also distorts your voice, so you always come up as 'Unknown' when speaking, even when you have another ID on. Obviously has anti-syngery with the Agent Card, but you're not wearing this to disguise as other people.
**Infiltrator Gloves:** Special combat gloves that allow you to fireman carry people more quickly.
**Sneakboots:** They muffle your footsteps completely!
**Carrying Case:** Storage container for the full suit and guns if you have them. Guns not included.

Screenshot 2020-02-28 06 08 03

ss+(2020-02-26+at+08 29 43)

Also includes the Sleepytime Pajama Bundle for nukies.

Zawo and I have been passing this between us for a while trying to get this into a good state. Hopefully now it's worth an inclusion.

## Why It's Good For The Game

~~A cheaper alternative to the Syndicate Balloon to indicate to everyone around you that you are probably valid.~~

A recognizable outfit that proves to be useful in minor ways and provide cheap armor for traitors. Nothing this grants is especially powerful on it's own, but it is a useful for traitors wanting to gain an edge at low cost.

## Changelog
:cl: NecromancerAnne and zawo
add: The Infiltrator Bundle, an armor kit for 3TC. Murder people in style!
add: Some pajamas for nukies to get plenty of bed rest.
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
No suspicious turtleskirt?

**Comment by Fikou** [comment]
Why It's Good For The Game?

**Comment by necromanceranne** [comment]
There is one just, I couldn't think of a way to put it in the uplink and limit the purchases without making it, iunno, a beacon. And a beacon drop podding in clothes for you will probably turn heads immediately before even the turtleneck.

**Comment by necromanceranne** [comment]
@Fikou Considering this is weaker than actually spending TC on the far more versatile and useful Chameleom Kit, which also happens to be armoured and actually significantly moreso than this turtleneck, I figured adding a silly one time purchase valid suit would be fairly tame. The alternative is always just getting a security jumpsuit, which are not at all difficult to acquire and have a level of deniability about them.

This has no amount of deniability and that's what makes it funny when it gets you killed.

**Comment by Mickyan** [comment]
What's funny about the syndicate balloon is that it costs all your TC this is just free armor on demand

**Comment by Tlaltecuhtli** [comment]
giving a free syndibox of clothes to every traitor is weird they are gonna just throw it in the hall and make a mess for sec

File: code/game/objects/items/storage/uplink_kits.dm [hunk 1]
```diff
@@ -530,6 +530,16 @@
 	for(var/i in 1 to 3)
 		new /obj/item/grenade/spawnergrenade/buzzkill(src)
 
+/obj/item/storage/box/syndie_kit/clothes
+	name = "box of clothing"
+	desc = "This clothes are...really red."
```
Comment (Fikou) [inline code/game/objects/items/storage/uplink_kits.dm]: only one clothing item is red
also a space should be after the dots

File: code/modules/clothing/under/syndicate.dm [hunk 2]
```diff
@@ -20,6 +20,22 @@
...
+
+/obj/item/clothing/under/syndicate/suspicious/skirt
+	name = "suspicious skirtleneck"
+	desc = "Gee, you think the person wearing this might be a member of a clandestine terrorist organization?"
+	icon_state = "suspicious_skirt"
+	item_state = "bl_suit"
+	has_sensor = NO_SENSORS
+	fitted = FEMALE_UNIFORM_TOP
+	can_adjust = FALSE
+
```
Comment (Fikou) [inline code/modules/clothing/under/syndicate.dm]: lots of this stuff is already inherited

**Comment by necromanceranne** [comment]
Tlal has a good point and it's just gonna get tossed in the halls. At least if it costs your murderdollars it at least isn't giving away the round-type/presence od antags at no charge.

I'll fix this in the morning and make it cost at least 1tc.

**Comment by improvedname** [comment]
Is the syndicate hardsuit in combat mode an joke to you?

**Comment by trollbreeder** [comment]
oh look, more uplink bloat

**Comment by optimumtact** [comment]
it's uplink bloat and I don't like it 👎

**Comment by itseasytosee** [comment]
Do not have free uplink items.

**Comment by Tharcoonvagh** [comment]
How so?

**Comment by HomocidlePancakes** [comment]
5 armor jumpsuit promotes murderboning *WHEEEEZE*

**Comment by Twaticus** [comment]
i remember some discussion about putting it in the blackmarket uplink and i kinda liked that idea as an alternative

**Comment by necromanceranne** [comment]
If the orange man doesn't like it for the actual uplink I might make an alternative to do just that.

**Comment by necromanceranne** [comment]
I've made a lot of changes working with another person to complete this idea. It's now a sneaksuit and worth the 1tc (I think).

Also pajamas. I'll add some fuzzy slippers maybe later.

**Comment by itseasytosee** [comment]
Also not even crediting me for syndie pajamas in #49458 smh.

**Comment by necromanceranne** [comment]
Now it includes Infiltrator Gloves, which are nitrile gloves but more valid and insulated. And sneakboots, which, if it works, should muffle your footsteps. 

I'll test this tomorrow, I need sleep.

**Comment by necromanceranne** [comment]
Well, this PR definitely has really changed over time. You can get the full lowdown on what the bundle has become in the full body of the PR.

**Comment by necromanceranne** [comment]
Also: available to nukies y/n?

**Comment by optimumtact** [comment]
i reckon u should make a new pnr so people dont' see out of date comments

**Comment by necromanceranne** [comment]
Yeah sure thing orangeman.

File: code/datums/components/footstep.dm [hunk 3]
```diff
@@ -94,11 +94,15 @@
 	var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))
 
 	if(H.shoes || feetCover) //are we wearing shoes
+		if(HAS_TRAIT(H, TRAIT_SILENT_FOOTSTEPS))
+			return
```
Comment (Rohesie) [inline code/datums/components/footstep.dm]: You could make these two checks as single one at the start of the proc instead.
Something like:
```
/datum/component/footstep/proc/play_humanstep()
  if(HAS_TRAIT(parent, TRAIT_SILENT_FOOTSTEPS))
    return
```

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -18,6 +18,24 @@
...
+	desc = "These boots have special noise cancelling soles. Perfect for stealth, if it wasn't for the color scheme."
+	icon_state = "sneakboots"
+	item_state = "sneakboots"
+	resistance_flags = FIRE_PROOF |  ACID_PROOF
+
+/obj/item/clothing/shoes/combat/sneakboots/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot != ITEM_SLOT_FEET)
+		return
+	if(slot == ITEM_SLOT_FEET)
```
Comment (Rohesie) [inline code/modules/clothing/shoes/miscellaneous.dm]: No need for this check. The previous one makes it certain to be truthy.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 5]
```diff
@@ -18,6 +18,24 @@
...
+		ADD_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
+
+/obj/item/clothing/shoes/combat/sneakboots/dropped(mob/living/carbon/human/user)
+	. = ..()
+	REMOVE_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
```
Comment (Rohesie) [inline code/modules/clothing/shoes/miscellaneous.dm]: You might want to perform the action first and later call parent with `return ..()` on removal/deleting procs in general, as parent calls may do some cleanup actions changing the state of the item.
As in, replace `. = ..()` in the beginning with `return ..()`

File: code/modules/clothing/suits/armor.dm [hunk 6]
```diff
@@ -172,6 +172,15 @@
...
 	allowed = GLOB.detective_vest_allowed
 
+/obj/item/clothing/suit/armor/vest/infiltrator
+	name = "infiltrator vest"
+	desc = "This vest appears to be made of of highly flexible materials that absorb impacts with ease."
+	icon_state = "infiltrator"
+	item_state = "infiltrator"
+	armor = list("melee" = 30, "bullet" = 40, "laser" = 20, "energy" = 30, "bomb" = 70, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	strip_delay = 80
```
Comment (Rohesie) [inline code/modules/clothing/suits/armor.dm]: I'm personally fan of time defines. `strip_delay = 8 SECONDS` looks much clearer to me.

File: code/modules/mob/living/carbon/human/say.dm [hunk 7]
```diff
@@ -16,6 +16,10 @@
 				return real_name
 		else
 			return real_name
+	if(istype(wear_mask, /obj/item/clothing/mask/infiltrator))
+		var/obj/item/clothing/mask/infiltrator/V = wear_mask
+		if(V.vunknown)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/say.dm]: This is a property pretty similar to the chameleon's `vchange` and could be generalized as a mask property.
As in moving both variables to the mask level (maybe renaming it to something more clear like `voice_change` and `voice_unknown`) and performing a single `istype` check here. More generic behavior.

**Pull request closed without merging**